playbooks/roles/kubernetes/tasks/deploy-zowe-k8.yaml

```diff
@@ -0,0 +1,310 @@
+      context: "{{ k8s_context }}"
+      state: present 
+      src: "{{ item.path }}"
+    with_items: "{{ k8s_ns.files }}"
+  - name: Git checkout unmodified file
+    delegate_to: localhost
+    command: git checkout -- ../containers/kubernetes/samples/workspace-pvc.yaml 
+    ignore_errors: True
+  - name: Edit workspace-pvc file to update storageclass name
+    delegate_to: localhost
+    yedit:
+      src: ../containers/kubernetes/samples/workspace-pvc.yaml
+      edits:
+      - key: spec.storageClassName
+        value: "{{ k8s_storageclass }}"
+    when: k8s_storageclass != none
+  - name: Edit workspace-pvc file to update pv label 
+    delegate_to: localhost
+    yedit:
+      src: ../containers/kubernetes/samples/workspace-pvc.yaml
+      edits:
+      - key: metadata.labels
+        value:
+          app.kubernetes.io/name: zowe
+          app.kubernetes.io/instance: zowe
+          app.kubernetes.io/managed-by: manual
+          billingType: "{{ k8s_pvc_labels[0] }}" 
+          region: "{{ k8s_pvc_labels[1] }}"
+          zone: "{{ k8s_pvc_labels[2] }}"
+    when: k8s_pvc_labels != none
+  - name: Edit workspace-pvc file to update pv name
+    delegate_to: localhost
+    yedit:
+      src: ../containers/kubernetes/samples/workspace-pvc.yaml
+      edits:
+      - key: spec
+        value: 
+          storageClassName: ""
+          volumeName: "{{ k8s_pv_name }}" 
+          accessModes:
+            - ReadWriteMany
+          resources:
+            requests:
+              storage: 50Mi
+    when: k8s_pv_name != none
+  - name: Deploy workspace-pvc manifest 
+    delegate_to: localhost
+    kubernetes.core.k8s:
+      kubeconfig: "{{ kubeconfig }}" 
+      context: "{{ k8s_context }}"
+      state: present 
+      src: ../containers/kubernetes/samples/workspace-pvc.yaml
+
+#  - name: Run the convert-for-k8s script to gather ConfigMap & Secret from z/OS system for loadbalancer service. 
+#    raw: >-
+#      {{ convert_for_k8s }} -q -x {{ k8s_list_domain_ip }}
+#    register: k8s_conf_secret
+#    when: k8s_service|lower == "loadbalancer"
+#  - name: Run the convert-for-k8s script to gather ConfigMap & Secret from z/OS system for nodeport service. 
+#    raw: >-
+#      {{ convert_for_k8s }} -q -x {{ k8s_list_domain_ip }} -e 32554
+#    register: k8s_conf_secret
+#    when: k8s_service|lower == "nodeport"
+#  - name: save the register config and secret to file
+#    delegate_to: localhost
+#    copy: 
+#      content: "{{ k8s_conf_secret.stdout }}"
+#      dest: config_secret.yaml 
+  - name: Verify if zowe_instance_dir variable is defined. 
+    fail: msg="Please provide zowe_instance_dir; it's required for to run convert_for_k8s script"
+    when: zowe_instance_dir is none 
+  - name: Run the convert-for-k8s script to gather ConfigMap & Secret from z/OS system for loadbalancer service. 
+    raw: >-
+      {{ zowe_instance_dir }}/bin/utils/convert-for-k8s.sh -q -x {{ k8s_list_domain_ip }} > /tmp/config_secret.yaml
```

Comment:
We can use `{{ work_dir_remote }}/config_secret.yaml` instead of hardcoded `/tmp/config_secret.yaml`. Sometimes tmp directory has free space issues.

Reply:
Okay.